cli/tests/pcluster/delete/test_pcluster_delete.py

```diff
@@ -0,0 +1,141 @@
+    mocker.patch("pcluster.commands.utils.warn")
+    mocker.patch("pcluster.commands.PclusterConfig.init_aws")
+    args = get_fake_pdelete_args(keep_logs=keep_logs)
+
+    if not stack_exists:
+        with pytest.raises(SystemExit) as sysexit:
+            commands.delete(args)
+        assert_that(sysexit.value.code).is_equal_to(0)
+    else:
+        commands.delete(args)
+
+    commands.PclusterConfig.init_aws.called_with(args.config_file)
+
+    assert_that(commands.utils.warn.call_count).is_equal_to(warn_call_count)
+    if warn_call_count > 0:
+        commands.utils.warn.assert_called_with("Cluster {0} has already been deleted.".format(args.cluster_name))
+
+    assert_that(commands._persist_cloudwatch_log_groups.called).is_equal_to(persist_called)
+    if persist_called:
+        commands._persist_cloudwatch_log_groups.assert_called_with(args.cluster_name)
+
+    assert_that(commands._delete_cluster.called).is_equal_to(delete_called)
+    if delete_called:
+        commands._delete_cluster.assert_called_with(args.cluster_name, args.nowait)
+
+
+@pytest.mark.parametrize(
+    "stacks,template",
+    [
+        ([], {},),
+        (
+            [{"StackName": FAKE_STACK_NAME}, {"StackName": "substack_one"}],
+            {"Resources": {"key": {"DeletionPolicy": "Retain"}}},
+        ),
+        ([{"StackName": FAKE_STACK_NAME}], {"Resources": {"key": {"DeletionPolicy": "Don't Retain"}}},),
+    ],
+)
+def test_persist_cloudwatch_log_groups(mocker, stacks, template):
+    """Verify that commands._persist_cloudwatch_log_groups behaves as expected."""
+    mocker.patch("pcluster.commands.utils.get_cluster_substacks").return_value = stacks
+    mocker.patch("pcluster.commands.utils.get_stack_template").return_value = template
+    mocker.patch("pcluster.commands._persist_cloudwatch_log_group")
+    if template.get("Resources", {}).get("key", {}).get("DeletionPolicy") == "Retain":
+        mocker.patch("pcluster.commands._get_unretained_cw_log_group_resource_keys").return_value = ["key"]
+        expected_persist_cloudwatch_log_groups_call_count = len(stacks)
+    else:
+        mocker.patch("pcluster.commands._get_unretained_cw_log_group_resource_keys").return_value = []
+        expected_persist_cloudwatch_log_groups_call_count = 0
+    commands._persist_cloudwatch_log_groups(FAKE_CLUSTER_NAME)
+    commands.utils.get_cluster_substacks.assert_called_with(FAKE_CLUSTER_NAME)
+    assert_that(commands.utils.get_stack_template.call_count).is_equal_to(len(stacks))
+    assert_that(commands._get_unretained_cw_log_group_resource_keys.call_count).is_equal_to(len(stacks))
+    assert_that(commands._persist_cloudwatch_log_group.call_count).is_equal_to(
+        expected_persist_cloudwatch_log_groups_call_count
+    )
+
+
+@pytest.mark.parametrize(
+    "template",
+    [
+        {},
+        {"Resources": {}},
+        {"Resources": {"key": {}}},
+        {"Resources": {"key": {"DeletionPolicy": "Don't Retain"}}},
```

Comment:
Minor: Don't Retain --> Delete?

Reply:
Same as the other response on this. Added a `Delete` case.